aws/data_source_aws_s3_bucket_object.go

```diff
@@ -185,7 +185,7 @@ func dataSourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) e
 	// The "STANDARD" (which is also the default) storage
 	// class when set would not be included in the results.
 	d.Set("storage_class", s3.StorageClassStandard)
-	if out.StorageClass != nil {
+	if out.StorageClass != nil { // nosem: helper-schema-ResourceData-Set-extraneous-nil-check
```

Comment:
I considered reversing the check here, i.e.:

```
if out.StorageClass == nil {
    d.Set("storage_class", s3.StorageClassStandard)
} else {
    d.Set("storage_class", out.StorageClass)
}
```
Personally, I find that makes the intent clearer but ignoring the lint also works for me.